src/models/User.ts

```diff
@@ -52,18 +54,40 @@ const PROFILE_FIELDS = [
   "locale",
 ];
 
+function parseStringToType<T>(schema: ZodSchema<T>, input?: string): T | null {
+  if (!input) {
+    return null;
+  }
+
+  return schema.parse(JSON.parse(input));
+}
+
 async function getProfile(storage: DurableObjectStorage) {
   const jsonData = await storage.get<string>(StorageKeys.profile);
 
-  try {
-    return ProfileSchema.parse(JSON.parse(jsonData as string));
-  } catch (err) {
-    return null;
-  }
+  return parseStringToType<Profile>(ProfileSchema, jsonData);
+}
+
+async function getPasswordResetCode(storage: DurableObjectStorage) {
+  const jsonData = await storage.get<string>(StorageKeys.passwordResetCode);
+
+  return parseStringToType<Code>(CodeSchema, jsonData);
+}
+
+async function getAuthentiationCode(storage: DurableObjectStorage) {
```

Comment:
Typo - should be authentication. If I change this will everything blow up? 😃

```suggestion
async function getAuthenticationCode(storage: DurableObjectStorage) {
```